lib/ui/painting.dart

```diff
@@ -5521,8 +5563,21 @@ class ImmutableBuffer extends NativeFieldWrapperClass1 {
       instance._init(list, callback);
     }).then((_) => instance);
   }
+
+  /// Create a buffer from the asset with key [assetKey].
+  ///
+  /// Throws an [Exception] if the asset does not exist.
+  static Future<ImmutableBuffer> fromAsset(String assetKey) {
+    final ImmutableBuffer instance = ImmutableBuffer._(0);
+    return _futurize((_Callback<void> callback) {
+      return instance._initFromAsset(assetKey, callback);
```

Comment:
_futurize handles throwing an exception if this method returns a non-null string.